packages/darklang/stdlib/list.dark

```diff
@@ -193,28 +193,22 @@ module Darklang =
       // results of those calls.
       // Consider <fn List.filterMap> if you also want to drop some of the values.
       let map (list: List<'a>) (fn: 'a -> 'b) : List<'b> =
-        match list with
-        | [] -> []
-        | head :: tail ->
-          PACKAGE.Darklang.Stdlib.List.push_v0
-            (PACKAGE.Darklang.Stdlib.List.map tail fn)
-            (fn head)
+        (PACKAGE.Darklang.Stdlib.List.fold list [] (fun acc elem ->
+          PACKAGE.Darklang.Stdlib.List.push_v0 acc (fn elem)
+        )) |> PACKAGE.Darklang.Stdlib.List.reverse
+
 
 
       // Calls <param f> on every <var val> in <param list>, returning a list of only
       // those values for which {{fn val}} returns {{true}}.
       // Preserves the order of values that were not dropped. Consider <fn
       // List.filterMap> if you also want to transform the values.
       let filter (list: List<'a>) (f: 'a -> Bool) : List<'a> =
-        match list with
-        | [] -> []
-        | head :: tail ->
-          if (head |> f) then
-            PACKAGE.Darklang.Stdlib.List.push_v0
-              (PACKAGE.Darklang.Stdlib.List.filter tail f)
-              head
+        (PACKAGE.Darklang.Stdlib.List.fold list [] (fun acc elem ->
```

Comment:
I believe this implementation is more performant than the previous one. wdyt?

Reply:
Is it worth renaming our List.fold to List.foldRight and add List.foldLeft?

Reply:
Maybe - is that what they are? I forget the difference between foldLeft and foldRight. I wouldn't mind having fold be a synonym of whichever one is appropriate as well.

Reply:
> I believe this implementation is more performant than the previous one. wdyt?

Honestly no idea.